qtfred/src/ui/dialogs/MissionSpecs/CustomStringsDialog.cpp

```diff
@@ -0,0 +1,241 @@
+	ui->stringsTableView->verticalHeader()->setVisible(false);
+	ui->stringsTableView->horizontalHeader()->setStretchLastSection(true);
+	ui->stringsTableView->setSortingEnabled(false);
+
+	// Make sure headers are not interactive
+	auto* hdr = ui->stringsTableView->horizontalHeader();
+	hdr->setSectionsClickable(false);        // no click/press behavior
+	hdr->setSortIndicatorShown(false);       // hide sort arrow
+	hdr->setHighlightSections(false);        // don�t change look when selected
+	hdr->setSectionsMovable(false);          // no drag-to-reorder columns
+	hdr->setFocusPolicy(Qt::NoFocus);   
+
+	// When a selection is made then load the editors
+	connect(ui->stringsTableView->selectionModel(),
+		&QItemSelectionModel::selectionChanged,
+		this,
+		[this](const QItemSelection&, const QItemSelection&) {
+			const auto idx = ui->stringsTableView->currentIndex();
+			loadRowIntoEditors(idx.isValid() ? idx.row() : -1);
+		});
+}
+
+void CustomStringsDialog::refreshTable()
+{
+	_tableModel->setRowCount(0);
+
+	const auto& rows = _model->items();
+	_tableModel->insertRows(0, static_cast<int>(rows.size()));
+	for (int i = 0; i < static_cast<int>(rows.size()); ++i) {
+		const auto& e = rows[static_cast<size_t>(i)];
+		auto* keyItem = new QStandardItem(QString::fromStdString(e.name));
+		auto* valItem = new QStandardItem(QString::fromStdString(e.value));
+		keyItem->setEditable(false);
+		valItem->setEditable(false);
+		_tableModel->setItem(i, ColKey, keyItem);
+		_tableModel->setItem(i, ColValue, valItem);
+	}
+}
+
+void CustomStringsDialog::selectRow(int row)
+{
+	if (row < 0 || row >= _tableModel->rowCount()) {
+		ui->stringsTableView->clearSelection();
+		loadRowIntoEditors(-1);
+		return;
+	}
+	ui->stringsTableView->selectRow(row);
+	loadRowIntoEditors(row);
+}
+
+void CustomStringsDialog::loadRowIntoEditors(int row)
+{
+	util::SignalBlockers blockers(this);
+
+	if (row < 0 || row >= _tableModel->rowCount()) {
+		clearEditors();
+		return;
+	}
+
+	const auto& e = _model->items()[static_cast<size_t>(row)];
+	ui->keyLineEdit->setText(QString::fromStdString(e.name));
+	ui->valueLineEdit->setText(QString::fromStdString(e.value));
+	ui->stringTextEdit->setPlainText(QString::fromStdString(e.text));
+}
+
+custom_string CustomStringsDialog::editorsToEntry() const
+{
+	custom_string e;
+	e.name = ui->keyLineEdit->text().toUtf8().constData();
+	e.value = ui->valueLineEdit->text().toUtf8().constData();
+	e.text = ui->stringTextEdit->toPlainText().toUtf8().constData();
+	return e;
+}
+
+void CustomStringsDialog::clearEditors()
+{
+	util::SignalBlockers blockers(this);
+	
+	ui->keyLineEdit->clear();
+	ui->valueLineEdit->clear();
+	ui->stringTextEdit->clear();
+}
+
+void CustomStringsDialog::on_addButton_clicked()
+{
+	auto e = editorsToEntry();
+	SCP_string err;
+	if (!_model->add(e, &err)) {
+		QMessageBox::warning(this, tr("Invalid Entry"), QString::fromStdString(err));
+		return;
+	}
+
+	refreshTable();
+	//selectRow(_tableModel->rowCount() - 1);
```

Comment:
?

Reply:
Yeah I dunno why this was commented out. It should be in. I tested to be sure. Fixed.